chaste_codegen/rl_model.py

```diff
@@ -0,0 +1,139 @@
+            beta = sp.Wild('beta', exclude=[x])
+            match = expr.expand().match((alpha - x * alpha) - beta * x)
+            if match is not None:
+                a, b = match[alpha], match[beta]
+            return {'alpha': a, 'beta': b}
+
+        def match_inf_tau(expr, x):  # expr already in piecewise_fold form
+            """Match (inf-x)/tau"""
+            i, t = None, None
+            inf = sp.Wild('inf', exclude=[x])
+            tau = sp.Wild('tau', exclude=[x])
+            match = expr.expand().match(inf / tau - x / tau)
+            if match is not None:
+                i, t = match[inf], match[tau]
+            return {'inf': i, 'tau': t}
+
+        derivative_alpha_beta, vars_in_derivative_alpha_beta = [], set()
+
+        # Substitute non-linear bits into derivative equations, so that we can pattern match
+        linear_derivs_eqs = derives_eqs_partial_eval_non_linear(self._y_derivatives, self._non_linear_state_vars,
+                                                                self._membrane_voltage_var, self._state_vars,
+                                                                self._get_equations_for)
+
+        for deriv in self._y_derivatives:
+            ab = {'alpha': None}
+            it = {'tau': None}
+            # get match if possible (deiv is linear)
+            if deriv.args[0] not in self._non_linear_state_vars and deriv.args[0] is not self._membrane_voltage_var:
+                eq = [e for e in linear_derivs_eqs if e.lhs == deriv][0]
+                ab = match_alpha_beta(eq.rhs, eq.lhs.args[0])
+                it = match_inf_tau(eq.rhs, eq.lhs.args[0])
+
+            # check if there was a match
+            if ab['alpha'] is not None or it['tau'] is not None:
+                eq = [e for e in linear_derivs_eqs if e.lhs == deriv][0]
+                rAlphaOrTau = ab['alpha'] if ab['alpha'] else it['tau']
+                rBetaOrInf = ab['beta'] if ab['beta'] else it['inf']
+                derivative_alpha_beta.append({'type': 'alphabeta' if ab['alpha'] else 'inftau',
+                                              'rAlphaOrTau': self._printer.doprint(rAlphaOrTau),
+                                              'rBetaOrInf': self._printer.doprint(rBetaOrInf)})
+                vars_in_derivative_alpha_beta.update(rAlphaOrTau.free_symbols | rBetaOrInf.free_symbols)
+            else:
+                derivative_alpha_beta.append({'type': 'non_linear', 'deriv': self._printer.doprint(deriv)})
+                vars_in_derivative_alpha_beta.add(deriv)
+
+        deriv_eqs_EvaluateEquations = self._get_equations_for(vars_in_derivative_alpha_beta)
+
+        # Get all used variables for derivative eqs
+        deriv_variables = set()
+        for eq in deriv_eqs_EvaluateEquations:
+            deriv_variables.update(eq.rhs.free_symbols)
+
+        for sv in self._formatted_state_vars:
+            sv['in_ab'] = sv['sympy_var'] in deriv_variables
+        # quicker way?
```

Comment:
?

Reply:
uups an old comment I did some elaborate slow way before and knew it wasn't efficient